internal/scheduler/common.go

```diff
@@ -0,0 +1,278 @@
+package scheduler
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/exp/maps"
+
+	"github.com/armadaproject/armada/internal/armada/configuration"
+	armadamaps "github.com/armadaproject/armada/internal/common/maps"
+	armadaresource "github.com/armadaproject/armada/internal/common/resource"
+	armadaslices "github.com/armadaproject/armada/internal/common/slices"
+	schedulerconfig "github.com/armadaproject/armada/internal/scheduler/configuration"
+	schedulercontext "github.com/armadaproject/armada/internal/scheduler/context"
+	"github.com/armadaproject/armada/internal/scheduler/interfaces"
+	"github.com/armadaproject/armada/internal/scheduler/schedulerobjects"
+)
+
+// SchedulerResult is returned by Rescheduler.Schedule().
+type SchedulerResult struct {
```

Comment:
would it be possible to define this as `type SchedulerResult[T interfaces.LegacySchedulerJob] struct`? That would mean:
- Functions like `NewSchedulerResult` would be simpler because you wouldn't need to do the casting
- You could remove PreemptedJobsFromSchedulerResult and the associated lack of compile-time safety
- Performance would be better because we wouldn't needs to reallocate arrays, just to perform casts.

Reply:
Yes, but it would mean adding type parameters to large parts of the scheduler. If we're anyway moving away from using this interface later, I'd prefer to not add a type parameter to `SchedulerResult` now.